### ISSUE TITLE
dnsmasq not starting

### ISSUE THREAD

**Issue by hobbypunk90** [hassio-addons/addon-pi-hole#35]
# Problem/Motivation

since the update this night pi-hole end in an infinity loop with this error:
`dnsmasq: unknown user or group: dnsmasq,`

## Steps to reproduce

update to 2.0.1

## Hardware

* raspberry pi 3

## Software

* Arch Linux aarch64
* Home Assistant 0.75.3
* Addon pi-hole 2.0.1

## Full Log
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.,
[s6-init] ensuring user provided files have correct perms...exited 0.,
[fix-attrs.d] applying ownership & permissions fixes...,
[fix-attrs.d] 01-resolver-resolv: applying... ,
[fix-attrs.d] 01-resolver-resolv: exited 0.,
[fix-attrs.d] 01-sudo: applying... ,
[fix-attrs.d] 01-sudo: exited 0.,
[fix-attrs.d] 02-nginx: applying... ,
[fix-attrs.d] 02-nginx: exited 0.,
[fix-attrs.d] 03-pihole: applying... ,
[fix-attrs.d] 03-pihole: exited 0.,
[fix-attrs.d] done.,
[cont-init.d] executing container initialization scripts...,
[cont-init.d] 00-banner.sh: executing... ,
,
-----------------------------------------------------------,
 Hass.io Add-on: Pi-hole v2.0.1,
,
 Network-wide ad blocking using your Hass.io instance,
,
 From: Community Hass.io Add-ons,
 By: Franck Nijhof <frenck@addons.community>,
-----------------------------------------------------------,
 aarch64 / Arch Linux ARM / HA 0.75.3 / SU 127 / stable,
-----------------------------------------------------------,
[cont-init.d] 00-banner.sh: exited 0.,
[cont-init.d] 01-log-level.sh: executing... ,
Log level is set to INFO,
[cont-init.d] 01-log-level.sh: exited 0.,
[cont-init.d] 02-updates.sh: executing... ,
,
INFO: You are running the latest version of this add-on,
[cont-init.d] 02-updates.sh: exited 0.,
[cont-init.d] 10-requirements.sh: executing... ,
INFO: Password is NOT in the Have I Been Pwned database! Nice!,
[cont-init.d] 10-requirements.sh: exited 0.,
[cont-init.d] 20-patches.sh: executing... ,
,
[cont-init.d] 20-patches.sh: exited 0.,
[cont-init.d] 21-v4-migration.sh: executing... ,
[cont-init.d] 21-v4-migration.sh: exited 0.,
[cont-init.d] 30-pihole.sh: executing... ,
[cont-init.d] 30-pihole.sh: exited 0.,
[cont-init.d] 31-pihole-ftl.sh: executing... ,
[cont-init.d] 31-pihole-ftl.sh: exited 0.,
[cont-init.d] 32-nginx.sh: executing... ,
[cont-init.d] 32-nginx.sh: exited 0.,
[cont-init.d] 33-password.sh: executing... ,
  [✓] New password set,
[cont-init.d] 33-password.sh: exited 0.,
[cont-init.d] 34-hostname.sh: executing... ,
,
[cont-init.d] 34-hostname.sh: exited 0.,
[cont-init.d] 35-logfiles.sh: executing... ,
[cont-init.d] 35-logfiles.sh: exited 0.,
[cont-init.d] 36-hosts.sh: executing... ,
[cont-init.d] 36-hosts.sh: exited 0.,
[cont-init.d] 50-gravity.sh: executing... ,
[cont-init.d] 50-gravity.sh: exited 0.,
[cont-init.d] 99-tests.sh: executing... ,
dnsmasq: syntax check OK.,
[24-Aug-2018 09:30:41] NOTICE: configuration file /etc/php7/php-fpm.conf test is successful,
,
nginx: the configuration file /etc/nginx/nginx.conf syntax is ok,
nginx: configuration file /etc/nginx/nginx.conf test is successful,
[cont-init.d] 99-tests.sh: exited 0.,
[cont-init.d] done.,
[services.d] starting services,
[services.d] done.,
,
dnsmasq: unknown user or group: dnsmasq,
,
dnsmasq: unknown user or group: dnsmasq,
,
dnsmasq: unknown user or group: dnsmasq,
,
dnsmasq: unknown user or group: dnsmasq,
,
dnsmasq: unknown user or group: dnsmasq,
,
dnsmasq: unknown user or group: dnsmasq,
,
dnsmasq: unknown user or group: dnsmasq,
,
dnsmasq: unknown user or group: dnsmasq,
,
dnsmasq: unknown user or group: dnsmasq,
,
dnsmasq: unknown user or group: dnsmasq,
,
dnsmasq: unknown user or group: dnsmasq,
,
dnsmasq: unknown user or group: dnsmasq,
,
dnsmasq: unknown user or group: dnsmasq,
,
dnsmasq: unknown user or group: dnsmasq,
,
dnsmasq: unknown user or group: dnsmasq,
,
dnsmasq: unknown user or group: dnsmasq,
[cont-finish.d] executing container finish scripts...,
[cont-finish.d] 99-message.sh: executing... ,
[cont-finish.d] 99-message.sh: exited 0.,
[cont-finish.d] done.,
[s6-finish] syncing disks.,
[s6-finish] sending all processes the TERM signal.,
```

**Comment by frenck** [comment]
Thank you for reporting this in!

I'll mark this issue critical priority.

**Comment by brentil** [comment]
I wanted to chime in with an upgrade from 1.3.0 to 2.0.2 I had the same problem.  I restarted the add-on and same thing happens.  Pi3, Arch Linux aarch64, HA 0.76.2, SU 127, HassIO 1.9. 

> [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
> [s6-init] ensuring user provided files have correct perms...exited 0.
> [fix-attrs.d] applying ownership & permissions fixes...
> [fix-attrs.d] 01-resolver-resolv: applying... 
> [fix-attrs.d] 01-resolver-resolv: exited 0.
> [fix-attrs.d] 01-sudo: applying... 
> [fix-attrs.d] 01-sudo: exited 0.
> [fix-attrs.d] 02-nginx: applying... 
> [fix-attrs.d] 02-nginx: exited 0.
> [fix-attrs.d] 03-pihole: applying... 
> [fix-attrs.d] 03-pihole: exited 0.
> [fix-attrs.d] done.
> [cont-init.d] executing container initialization scripts...
> [cont-init.d] 00-banner.sh: executing... 
> -----------------------------------------------------------
>  Hass.io Add-on: Pi-hole v2.0.2
>  Network-wide ad blocking using your Hass.io instance
>  From: Community Hass.io Add-ons
>  By: Franck Nijhof <frenck@addons.community>
> -----------------------------------------------------------
>  aarch64 / HassOS 1.9 / HA 0.76.2 / SU 127 / stable
> -----------------------------------------------------------
> [cont-init.d] 00-banner.sh: exited 0.
> [cont-init.d] 01-log-level.sh: executing... 
> Log level is set to INFO
> [cont-init.d] 01-log-level.sh: exited 0.
> [cont-init.d] 02-updates.sh: executing... 
> INFO: You are running the latest version of this add-on
> [cont-init.d] 02-updates.sh: exited 0.
> [cont-init.d] 10-requirements.sh: executing... 
> INFO: Password is NOT in the Have I Been Pwned database! Nice!
> [cont-init.d] 10-requirements.sh: exited 0.
> [cont-init.d] 20-patches.sh: executing... 
> [cont-init.d] 20-patches.sh: exited 0.
> [cont-init.d] 21-v4-migration.sh: executing... 
> [cont-init.d] 21-v4-migration.sh: exited 0.
> [cont-init.d] 30-pihole.sh: executing... 
> [cont-init.d] 31-pihole-ftl.sh: executing... 
> [cont-init.d] 31-pihole-ftl.sh: exited 0.
> [cont-init.d] 32-nginx.sh: executing... 
> [cont-init.d] 32-nginx.sh: exited 0.
> [cont-init.d] 33-password.sh: executing... 
>   [✓] New password set
> [cont-init.d] 33-password.sh: exited 0.
> [cont-init.d] 34-hostname.sh: executing... 
> [cont-init.d] 34-hostname.sh: exited 0.
> [cont-init.d] 35-logfiles.sh: executing... 
> [cont-init.d] 35-logfiles.sh: exited 0.
> [cont-init.d] 36-hosts.sh: executing... 
> [cont-init.d] 36-hosts.sh: exited 0.
> [cont-init.d] 50-gravity.sh: executing... 
> [cont-init.d] 50-gravity.sh: exited 0.
> [cont-init.d] 99-tests.sh: executing... 
> dnsmasq: syntax check OK.
> [25-Aug-2018 15:52:53] NOTICE: configuration file /etc/php7/php-fpm.conf test is successful
> nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
> nginx: configuration file /etc/nginx/nginx.conf test is successful
> [cont-init.d] 99-tests.sh: exited 0.
> [cont-init.d] done.
> [services.d] starting services
> [services.d] done.
> dnsmasq: unknown user or group: dnsmasq
> dnsmasq: unknown user or group: dnsmasq
> dnsmasq: unknown user or group: dnsmasq
> dnsmasq: unknown user or group: dnsmasq
> dnsmasq: unknown user or group: dnsmasq
> dnsmasq: unknown user or group: dnsmasq
> dnsmasq: unknown user or group: dnsmasq

I can login to the admin interface still and the pihole-FTL.log shows the service cycling over and over again too.  The info below keeps rolling by every couple of seconds.

> [2018-08-25 16:04:47.700] ########## FTL started! ##########
> [2018-08-25 16:04:47.700] FTL branch: master
> [2018-08-25 16:04:47.700] FTL version: v4.0
> [2018-08-25 16:04:47.700] FTL commit: 8493df4-dirty
> [2018-08-25 16:04:47.701] FTL date: 2018-08-05 13:40:30 -0700
> [2018-08-25 16:04:47.701] FTL user: pihole
> [2018-08-25 16:04:47.701] Starting config file parsing (/etc/pihole/pihole-FTL.conf)
> [2018-08-25 16:04:47.701]    SOCKET_LISTENING: only local
> [2018-08-25 16:04:47.701]    AAAA_QUERY_ANALYSIS: Show AAAA queries
> [2018-08-25 16:04:47.702]    MAXDBDAYS: max age for stored queries is 365 days
> [2018-08-25 16:04:47.702]    RESOLVE_IPV6: Resolve IPv6 addresses
> [2018-08-25 16:04:47.702]    RESOLVE_IPV4: Resolve IPv4 addresses
> [2018-08-25 16:04:47.702]    DBINTERVAL: saving to DB file every minute
> [2018-08-25 16:04:47.702]    DBFILE: Using /etc/pihole/pihole-FTL.db
> [2018-08-25 16:04:47.703]    MAXLOGAGE: Importing up to 24.0 hours of log data
> [2018-08-25 16:04:47.703]    PRIVACYLEVEL: Set to 0
> [2018-08-25 16:04:47.703]    IGNORE_LOCALHOST: Show queries from localhost
> [2018-08-25 16:04:47.704]    BLOCKINGMODE: Null IPs for blocked domains
> [2018-08-25 16:04:47.704]    REGEX_DEBUGMODE: Inactive
> [2018-08-25 16:04:47.704] Finished config file parsing
> [2018-08-25 16:04:47.704] INFO: No Regex file found
> [2018-08-25 16:04:47.707] Database successfully initialized
> [2018-08-25 16:04:47.709] Notice: Increasing queries struct size from 0 to 10000
> [2018-08-25 16:04:47.709] Notice: Increasing domains struct size from 0 to 1000
> [2018-08-25 16:04:47.710] Notice: Increasing clients struct size from 0 to 10
> [2018-08-25 16:04:47.710] New forward server: 208.67.220.220 (0/0)
> [2018-08-25 16:04:47.710] Notice: Increasing forwarded struct size from 0 to 4
> [2018-08-25 16:04:47.710] Notice: Increasing overTime struct size from 0 to 100
> [2018-08-25 16:04:47.711] New forward server: 208.67.222.222 (1/4)
> [2018-08-25 16:04:47.771] Notice: Increasing overTime struct size from 100 to 200
> [2018-08-25 16:04:47.853] Notice: Increasing domains struct size from 1000 to 2000
> [2018-08-25 16:04:47.860] Imported 9157 queries from the long-term database
> [2018-08-25 16:04:47.861]  -> Total DNS queries: 9157
> [2018-08-25 16:04:47.861]  -> Cached DNS queries: 604
> [2018-08-25 16:04:47.861]  -> Forwarded DNS queries: 8504
> [2018-08-25 16:04:47.861]  -> Exactly blocked DNS queries: 49
> [2018-08-25 16:04:47.861]  -> Unknown DNS queries: 0
> [2018-08-25 16:04:47.862]  -> Unique domains: 1043
> [2018-08-25 16:04:47.862]  -> Unique clients: 7
> [2018-08-25 16:04:47.862]  -> Known forward destinations: 2
> [2018-08-25 16:04:47.862] Successfully accessed setupVars.conf
> [2018-08-25 16:04:50.003] ########## FTL started! ##########
> [2018-08-25 16:04:50.003] FTL branch: master
> [2018-08-25 16:04:50.003] FTL version: v4.0
> [2018-08-25 16:04:50.003] FTL commit: 8493df4-dirty
> [2018-08-25 16:04:50.003] FTL date: 2018-08-05 13:40:30 -0700
> [2018-08-25 16:04:50.003] FTL user: pihole

**Comment by highground88** [comment]
My experience is the same as the user above -

**Comment by frenck** [comment]
@highground88 Could you please share your logs?

They contain system information, which might be useful in locating the source of the issue.

**Comment by highground88** [comment]
.... and uninstalling and reinstalling with the same config script fixes the problem.

**Comment by brentil** [comment]
I have not reinstalled mine yet but I have it disabled for now.  This install of PiHole is actually my backup PiHole so when I'm doing maintenance on the primary the house always has DNS.

Not sure if it helps but this is my config.  I have 35 hosts entries in the list and with my added custom lists I have 500k domains in blocklist.

```
{
  "log_level": "info",
  "password": "XXXXXXXXXX",
  "update_lists_on_start": false,
  "admin_port": 4865,
  "dns_port": 53,
  "ssl": false,
  "certfile": "fullchain.pem",
  "keyfile": "privkey.pem",
  "interface": "",
  "ipv6": true,
  "ipv4_address": "",
  "ipv6_address": "",
  "virtual_host": "my.domain.local",
  "hosts": [
    {
      "name": "System01",
      "ip": "192.168.1.1"
    },
    {
      "name": "System02",
      "ip": "192.168.1.2"
    },
...
  ]
}
```

**Comment by brentil** [comment]
I turned the logging level up to debug but unfortunately once it hits the last section and the account starts cycling it filled up the logs too quickly for me to capture what was before the FTL failures.  It does show at least where in the steps of loading the dnsmasq error is popping out, which is in the same place in each iteration.

```
[2018-08-26 11:10:27.912] ########## FTL started! ##########
[2018-08-26 11:10:27.913] FTL branch: master
[2018-08-26 11:10:27.913] FTL version: v4.0
[2018-08-26 11:10:27.913] FTL commit: 8493df4-dirty
[2018-08-26 11:10:27.914] FTL date: 2018-08-05 13:40:30 -0700
[2018-08-26 11:10:27.914] FTL user: pihole
[2018-08-26 11:10:27.914] Starting config file parsing (/etc/pihole/pihole-FTL.conf)
[2018-08-26 11:10:27.914]    SOCKET_LISTENING: only local
[2018-08-26 11:10:27.914]    AAAA_QUERY_ANALYSIS: Show AAAA queries
[2018-08-26 11:10:27.915]    MAXDBDAYS: max age for stored queries is 365 days
[2018-08-26 11:10:27.915]    RESOLVE_IPV6: Resolve IPv6 addresses
[2018-08-26 11:10:27.915]    RESOLVE_IPV4: Resolve IPv4 addresses
[2018-08-26 11:10:27.915]    DBINTERVAL: saving to DB file every minute
[2018-08-26 11:10:27.916]    DBFILE: Using /etc/pihole/pihole-FTL.db
[2018-08-26 11:10:27.916]    MAXLOGAGE: Importing up to 24.0 hours of log data
[2018-08-26 11:10:27.916]    PRIVACYLEVEL: Set to 0
[2018-08-26 11:10:27.916]    IGNORE_LOCALHOST: Show queries from localhost
[2018-08-26 11:10:27.917]    BLOCKINGMODE: Null IPs for blocked domains
[2018-08-26 11:10:27.917]    REGEX_DEBUGMODE: Inactive
[2018-08-26 11:10:27.917] Finished config file parsing
[2018-08-26 11:10:27.917] INFO: No Regex file found
[2018-08-26 11:10:27.920] Database successfully initialized
[2018-08-26 11:10:27.921] SELECT * FROM queries WHERE timestamp >= 1535209827
[2018-08-26 11:10:27.923] Notice: Increasing queries struct size from 0 to 10000
[2018-08-26 11:10:27.923] Notice: Increasing domains struct size from 0 to 1000
[2018-08-26 11:10:27.923] Notice: Increasing clients struct size from 0 to 10
[2018-08-26 11:10:27.923] New forward server: 208.67.220.220 (0/0)
[2018-08-26 11:10:27.924] Notice: Increasing forwarded struct size from 0 to 4
[2018-08-26 11:10:27.924] Notice: Increasing overTime struct size from 0 to 100
[2018-08-26 11:10:27.926] New forward server: 208.67.222.222 (1/4)
[2018-08-26 11:10:27.956] Imported 2044 queries from the long-term database
[2018-08-26 11:10:27.957]  -> Total DNS queries: 2044
**_dnsmasq: unknown user or group: dnsmasq_**
[2018-08-26 11:10:27.957]  -> Cached DNS queries: 132
[2018-08-26 11:10:27.957]  -> Forwarded DNS queries: 1899
[2018-08-26 11:10:27.957]  -> Exactly blocked DNS queries: 13
[2018-08-26 11:10:27.957]  -> Unknown DNS queries: 0
[2018-08-26 11:10:27.958]  -> Unique domains: 568
[2018-08-26 11:10:27.958]  -> Unique clients: 5
[2018-08-26 11:10:27.958]  -> Known forward destinations: 2
[2018-08-26 11:10:27.958] Successfully accessed setupVars.conf
[2018-08-26 11:10:30.081] ########## FTL started! ##########
```

**Comment by JeroenTielen** [comment]
I have the same issue. Admin page is not available. So i dont want to reinstall as this would delwte mu whitelist probably.

**Comment by brentil** [comment]
@JeroenTielen You can export that configuration by going to PiHole -> Admin -> Settings -> Teleporter -> Teleporter Export and then Export.  This will export your Whitelist, Blacklist, and Regex filters which you can then re-import.  However!... This does not export your list of custom Blocklists so you will have to record those and then manually re-add each one.

**Comment by mjoshd** [comment]
Running the latest Hass.io on HassOS (rPi3 x64) and I also had the "dnsmasq: unknown user or group: dnsmasq" issue when trying to upgrade... tried multiple times with no luck. It was easy to restore Pi-hole between attempts with a snapshot I had made earlier. Eventually I gave up and did a fresh install of the add-on.. works perfect now.

**Comment by vermi5** [comment]
What I can't fathom is the incredible mess of a host system I must have so that if I try to uninstall the addon, Ubuntu cannot resolve URLs anymore (the homeassistant container still can resolve but If I press Install or update buttons on the frontend they do nothing) so I can't use the reinstall workaround.

**Comment by frenck** [comment]
I've located the issue, caused by trying to fix the issue on v2.0.1, which actually resulted in making it worse.

Created a change and will be dropping a new release asap.

Sorry about the inconvenience this has caused. Greetings from a holiday location abroad (thank god for the 4G mobile internet).
